### PR TITLE
Apply keyboard shortcut conversions between paragraphs and headings to all editor instances

### PR DESCRIPTION
**Pull request by t-hamano**
Follow-up on: #44681

## What?
This PR adds conversion keyboard shortcuts between paragraphs and headings to all editors, in addition to the post editor.

## How?

In addition to adding shortcuts, the following actions have been taken:

- Add E2E test
- Fix a problem with text alignment not being carried over

## Testing Instructions

In the post/edit-site/edit-widgets/customize-widgets, check the following:

- Should convert from a paragraph to a heading with Ctrl + Alt (Option) + [1-6]
- Should convert from a heading to a paragraph with Ctrl + Alt (Option) + 0
- Should change heading level with Ctrl + Alt (Option) + [1-6]
- Text alignment attribute should be inherited
- Two new items have been added to the keyboard shortcut modal

### CODE REVIEW

File: packages/edit-post/src/components/keyboard-shortcuts/index.js [hunk 1]
```diff
@@ -181,7 +184,7 @@ function KeyboardShortcuts() {
 		} );
 
 		registerShortcut( {
-			name: `core/block-editor/transform-heading-to-paragraph`,
+			name: `core/edit-post/transform-heading-to-paragraph`,
```
Comment (t-hamano) [inline packages/edit-post/src/components/keyboard-shortcuts/index.js:187]: I think the namespace should be unified.

File: packages/edit-post/src/components/keyboard-shortcuts/index.js [hunk 2]
```diff
@@ -70,15 +70,18 @@ function KeyboardShortcuts() {
...
+			destinationBlockName === 'core/paragraph' ? 'align' : 'textAlign';
+
 		replaceBlocks(
 			currentClientId,
 			createBlock( destinationBlockName, {
 				level,
-				content: currentContent,
-				align: currentAlign,
+				content: attributes.content,
+				...{ [ destinationTextAlign ]: attributes[ textAlign ] },
```
Comment (t-hamano) [inline packages/edit-post/src/components/keyboard-shortcuts/index.js:84]: Heading block and paragraph block have different property name for text alignment:

- Heading Block: `textAlign`
- Paragraph Block: `align`

Therefore, property name and value must be properly mapped.

File: test/e2e/specs/editor/blocks/heading.spec.js [hunk 3]
```diff
@@ -179,4 +179,95 @@ test.describe( 'Heading', () => {
...
+		// Focus the block content
+		await page.keyboard.press( 'Tab' );
+
+		await pageUtils.pressKeyWithModifier( 'access', '0' );
+
+		// Somehow, if we use toMatchObject, it fails with GitHub Actions.
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:paragraph {\"align\":\"center\"} -->
+<p class=\"has-text-align-center\">Heading</p>
+<!-- /wp:paragraph -->`
```
Comment (t-hamano) [inline test/e2e/specs/editor/blocks/heading.spec.js]: Very strange, but only this test fails if `toMatchObject` is used. But locally it always succeeds...

Comment (kevin940726) [reply test/e2e/specs/editor/blocks/heading.spec.js]: Oops! That's my bad! I recently changed the implementation of `editor.getBlocks` and it's causing this error. Rebasing should be able to reproduce this error. I'll put up another PR to fix that though! 🙇

Comment (kevin940726) [reply test/e2e/specs/editor/blocks/heading.spec.js]: Proposed a fix in https://github.com/WordPress/gutenberg/pull/48093.

Comment (t-hamano) [reply test/e2e/specs/editor/blocks/heading.spec.js]: Thanks for the investigation! I am checking to see if the test by toMatchObject passes correctly after temporarily applying the changes in #48093 to this PR.

**Review by kevin940726 — APPROVED** [review]
Can't speak for the code, but the tests look good 👍. We might need a rebase though.

**Review by draganescu — APPROVED** [review]
I think this is a great default functionality to have across all editors. Its kinda weird that it has to be spelled out three different times but 🤷🏻

**Pull request merged**